apiserver/facades/client/annotations/annotations.go

```diff
@@ -96,26 +105,65 @@ func (api *API) getEntityAnnotations(ctx context.Context, entityTag string) (map
 		return nil, errors.Trace(err)
 	}
 
-	id, err := annotations.ConvertTagToID(tag)
-	if err != nil {
-		return nil, errors.Trace(err)
-	}
+	switch tag.Kind() {
+	case names.CharmTagKind:
```

Comment:
This switch should be in the service, but I'm not going to re-write the whole of annotations here.